src/ContentAwareGenerator.php

```diff
@@ -98,8 +100,9 @@ public function generate($name, $parameters = [], $absolute = UrlGeneratorInterf
         }
 
         $this->unsetLocaleIfNotNeeded($route, $parameters);
+        $parameters[RouteObjectInterface::ROUTE_OBJECT] = $route;
 
-        return parent::generate($route, $parameters, $absolute);
+        return parent::generate(RouteObjectInterface::OBJECT_BASED_ROUTE_NAME, $parameters, $absolute);
```

Comment:
we need to switch this around. this is inheritance, not composition, so there should not be any unexpected side effects as with the chain router.